api/working-with-extensions/continuous-integration.md

```diff
@@ -95,3 +95,59 @@ script:
     yarn && yarn compile && yarn test
 cache: yarn
 ```
+
+## Automated publishing
+
+You can configure the CI to publish a new version of the extension automatically.
+
+The publish command is similar to the command of publishing from local environment using [`vsce`](https://github.com/Microsoft/vsce) service but this time the command needs to pass also the the Personal Access Token ("PAT").
+
+So, to configure the CI to also publish the extension, follow those steps:
+
+1. Install `vsce` as `devDependencies` (`npm install vsce --save-dev` or `yarn add vsce --dev`)
+2. Add a `deploy` script in `package.json`
+
+```json
+"scripts": {
+  "deploy": "vsce publish -p [your-personal-access-token]"
+}
+```
+
+3. Add the publish step in `azure-pipelines.yml`
+```yaml
+- bash: |
+    echo ">>> Publish"
+    yarn deploy
+  displayName: Publish
+  condition: and(succeeded(), eq(variables['Build.SourceBranch'], 'refs/heads/master'), eq(variables['Agent.OS'], 'Linux'))
+```
+
+You can use the [`condition`](https://docs.microsoft.com/azure/devops/pipelines/process/conditions) to tell the CI not to publish in certain cases. For example, publishing only if the tests are passed, only for `master` builds or only when the build is running on a specific platform.
+
+### Publishing a public project
+
+In case that the project is [public](https://docs.microsoft.com/azure/devops/pipelines/repos/github#public-github-repository-and-azure-pipelines-in-a-public-project) the PAT probably shouldn't be exposed with the rest of the code.
+
+In this case, you can store the PAT in a "secret variable". The value of that variable will not be exposed and you can use it in the `azure-pipelines.yml` file.
+
+To create a secret variable, follow the [instructions](https://docs.microsoft.com/azure/devops/pipelines/process/variables?tabs=classic%2Cbatch#secret-variables)
+
+Next steps will be:
+
+1. Declare the `deploy` script in `package.json` without the PAT
+
+```json
+"scripts": {
+  "deploy": "vsce publish -p"
+}
+```
+
+2. Use it in the `publish` step in `azure-pipelines.yml`
+
+```yaml
+- bash: |
+    echo ">>> Publish"
+    yarn deploy $([the-name-of-the-secret-variable])
+  displayName: Publish
+  condition: and(succeeded(), eq(variables['Build.SourceBranch'], 'refs/heads/master'), eq(variables['Agent.OS'], 'Linux'))
```

Comment:
We should explain why `eq(variables['Agent.OS'], 'Linux')` is here and when one might want to remove it (if one only has one agent).

Reply:
Also, I believe something like `eq(variables['Build.SourceBranch'], 'refs/heads/master')` is wrong, since it would simply attempt to publish every single push. A better approach is to publish whenever a tag gets pushed to the repo. We can do this by first adding triggers to the yaml:

```yaml
trigger:
  branches:
    include: ['*']
  tags:
    include: ['*']
```

And then the condition would be:

```
  condition: and(succeeded(), startsWith(variables['Build.SourceBranch'], 'refs/tags/'))
```

Reply:
Also, I believe something like `eq(variables['Build.SourceBranch'], 'refs/heads/master')` is wrong, since it would simply attempt to publish every single push.

A better approach is to publish whenever a tag gets pushed to the repo. We can do this by first adding triggers to the yaml:

```yaml
trigger:
  branches:
    include: ['*']
  tags:
    include: ['*']
```

And then the condition would be:

```
  condition: and(succeeded(), startsWith(variables['Build.SourceBranch'], 'refs/tags/'))
```